porch/pkg/git/package.go

```diff
@@ -55,28 +56,36 @@ var _ repository.PackageRevision = &gitPackageRevision{}
 // layer, the prefix will be removed (this may happen without notice) so it should not
 // be relied upon by clients.
 func (p *gitPackageRevision) KubeObjectName() string {
-	hash := sha1.Sum([]byte(fmt.Sprintf("%s:%s:%s", p.repo.name, p.path, p.revision)))
+	// The published package revisions on the main branch will have the same description
+	// as the most recently published package revision, so we need to ensure it has a unique
+	// and unchanging name.
+	var s string
+	if p.revision == string(p.repo.branch) {
+		s = p.revision
+	} else {
+		s = p.description
+	}
+	hash := sha1.Sum([]byte(fmt.Sprintf("%s:%s:%s", p.repo.name, p.path, s)))
 	return p.repo.name + "-" + hex.EncodeToString(hash[:])
 }
 
 func (p *gitPackageRevision) Key() repository.PackageRevisionKey {
 	return repository.PackageRevisionKey{
-		Repository: p.repo.name,
-		Package:    p.path,
-		Revision:   p.revision,
+		Repository:  p.repo.name,
+		Package:     p.path,
+		Revision:    p.revision,
+		Description: p.description,
 	}
 }
 
 func (p *gitPackageRevision) uid() types.UID {
```

Comment:
Does this need the same logic as `KubeObjectName` to make sure it is unique for packagerevisions created before this change and therefore can have an empty description?

Reply:
I think it needs the same logic as `KubeObjectName` to make sure that packages on the main branch have a unique uid (since the description on `main` would be the same as the description of the latest publish revision) - I'll make that change. 

But I don't think the description field would ever be empty; everywhere that a `gitPackageRevision` is initialized, I initialize the description to be the same as the revision if it would otherwise be empty. I don't think we should be letting any package descriptions be empty, it would complicate a lot of the other checks, so this seemed like a reasonable default.

Reply:
Added the logic to `uid`